File: frontend/src/lib/api/maintenance.ts
import { apiClient } from './client'

export interface MaintenanceRequest {
  id: string
  property_id: string
  unit_id: string
  room_id?: string
  title: string
  description: string
  priority: 'LOW' | 'MEDIUM' | 'HIGH' | 'URGENT'
  status: 'open' | 'in_progress' | 'resolved' | 'closed'
  assigned_to?: string
  created_at: string
  resolved_at?: string
}

export interface MaintenanceWithDetails extends MaintenanceRequest {
  property_name: string
  room_number: string
  unit_number: string
  reporter_email: string
}

export const maintenanceApi = {
  getByProperty: async (propertyId: string): Promise<MaintenanceWithDetails[]> => {
    const { data } = await apiClient.get<MaintenanceWithDetails[]>(
      `/maintenance/property/${propertyId}`
    )
    return data
  },

  create: async (request: {
    property_id: string
    unit_id: string
    room_id: string | null
    title: string
    description: string
    priority: string
  }): Promise<MaintenanceRequest> => {
    const { data } = await apiClient.post<MaintenanceRequest>('/maintenance/', request)
    return data
  },

  update: async (
    id: string,
    updates: Partial<MaintenanceRequest>
  ): Promise<MaintenanceRequest> => {
    const { data } = await apiClient.put<MaintenanceRequest>(`/maintenance/${id}`, updates)
    return data
  },

  delete: async (id: string): Promise<void> => {
    await apiClient.delete(`/maintenance/${id}`)
  },
}

File: backend/app/models/maintenance.py
from sqlalchemy import Column, String, ForeignKey, Text, Enum as SQLEnum, ARRAY, DateTime
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import relationship
import enum

from app.models.base import BaseModel


class MaintenancePriority(str, enum.Enum):
    LOW = "LOW"
    MEDIUM = "MEDIUM"
    HIGH = "HIGH"
    URGENT = "URGENT"


class MaintenanceStatus(str, enum.Enum):
    OPEN = "open"
    IN_PROGRESS = "in_progress"
    RESOLVED = "resolved"
    CLOSED = "closed"


class MaintenanceRequest(BaseModel):
    __tablename__ = "maintenance_requests"
    
    property_id = Column(UUID(as_uuid=True), ForeignKey("properties.id"), nullable=False)
    unit_id = Column(UUID(as_uuid=True), ForeignKey("units.id"), nullable=False)
    room_id = Column(UUID(as_uuid=True), ForeignKey("rooms.id"))  # Nullable for common areas
    tenant_id = Column(UUID(as_uuid=True), ForeignKey("tenants.id"))
    title = Column(String(255), nullable=False)
    description = Column(Text)
    priority = Column(SQLEnum(MaintenancePriority), nullable=False, default=MaintenancePriority.MEDIUM)
    status = Column(SQLEnum(MaintenanceStatus), nullable=False, default=MaintenanceStatus.OPEN)
    photos = Column(ARRAY(Text))  # Array of S3 URLs
    assigned_to = Column(UUID(as_uuid=True), ForeignKey("users.id"))
    resolved_at = Column(DateTime(timezone=True))
    
    # Relationships
    property = relationship("Property", back_populates="maintenance_requests")
